src/number_theory/legendre_symbol/quadratic_char.lean

```diff
@@ -129,16 +124,33 @@ begin
         simp only [ha', h, neg_neg, eq_self_iff_true, if_true, if_false], }, },
 end
 
-/-- The quadratic character is a homomorphism of monoids with zero. -/
-@[simps] def quadratic_char_hom : F →*₀ ℤ :=
-{ to_fun := quadratic_char F,
-  map_zero' := quadratic_char_zero,
+variables (F)
+
+/-- The quadratic character as a multiplicative character. -/
+@[simps] def quadratic_char : mul_char F ℤ :=
+{ to_fun := quadratic_char' F,
   map_one' := quadratic_char_one,
-  map_mul' := quadratic_char_mul }
+  map_mul' := quadratic_char_mul,
+  map_nonunit' := λ a ha, by { rw of_not_not (mt ne.is_unit ha), exact quadratic_char_zero, } }
+
+variables {F}
+
+/-- For nonzero `a : F`, `quadratic_char F a = 1 ↔ is_square a`. -/
+lemma quadratic_char_one_iff_is_square {a : F} (ha : a ≠ 0) :
+  quadratic_char F a = 1 ↔ is_square a :=
+by { change quadratic_char' F a = 1 ↔ _,
```

Comment:
Two comments:
* you can add a lemma that says what you want instead of using `change` (the proof will be `rfl`), or you can state the lemma you need for `quadratic_char` instead of `quadratic_char'`.
* This proof is sufficiently long to have it with `begin... end`.

Reply:
It turns out that I can do without `change`, by adding `quadratic_char_apply` to the simp lemma list.